From 9553db1be43d52227127114b2daaef1baa881fcb Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:29 -0500
Subject: [drm] i915: Disable RCS flips on Ivybridge

Message-id: <1418852380-13061-19-git-send-email-rclark@redhat.com>
Patchwork-id: 102156
O-Subject: [RHEL7 drm 18/89] drm/i915: Disable RCS flips on Ivybridge
Bugzilla: 1153301
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1153301
Upstream: since v3.16.4

commit 0b44ffc73efd56f7c07276252a0cbf94dbc2fb78

Author:     Chris Wilson <chris@chris-wilson.co.uk>
AuthorDate: Tue Jul 8 10:40:29 2014 +0100
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:13:08 2014 -0500

    drm/i915: Disable RCS flips on Ivybridge

    commit 2a92d5bca1999b69c78f3c3e97b5484985b094b9 upstream.

    We currently see random GPU hangs when using RCS flips with multiple
    pipes on Ivybridge. Now that we have mmio flips, we can fairly cheaply
    fallback to using CPU driven flips instead.

    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=77104
    Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 437afa3..ebfec26 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -9315,6 +9315,8 @@ static int intel_crtc_page_flip(struct drm_crtc *crtc,
 
  if (IS_VALLEYVIEW(dev)) {
   ring = &dev_priv->ring[BCS];
+ } else if (IS_IVYBRIDGE(dev)) {
+  ring = &dev_priv->ring[BCS];
  } else if (INTEL_INFO(dev)->gen >= 7) {
   ring = obj->ring;
   if (ring == NULL || ring->id != RCS)
-- 
1.7.1